### PR TITLE
Improving scheduler extender unit tests

### PR DESCRIPTION
**Pull request by sathyanarays**
Signed-off-by: Sathyanarayanan Saravanamuthu <sathyanarays@blr-ubu-eng-037.vmware.com>

#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Improve the coverage of scheduler extender

#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:
This is my first PR. So, keeping the changes minimal to gain the PR experience.

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by kerthcet** [comment]
Hi @sathyanarays I think you should sign the CLA correctly at first.

**Comment by sathyanarays** [comment]
> Hi @sathyanarays I think you should sign the CLA correctly at first.

Yes. I am trying to do that :)

File: pkg/scheduler/testing/wrappers.go [hunk 1]
```diff
@@ -182,7 +182,18 @@ func (c *ContainerWrapper) Resources(resMap map[v1.ResourceName]string) *Contain
 	}
 	c.Container.Resources = v1.ResourceRequirements{
 		Requests: res,
-		Limits:   res,
```
Comment (kerthcet) [inline pkg/scheduler/testing/wrappers.go:185]: `Resources` is directly/indirectly invoked by lots of places, let's keep it the same. Add another method `ResourceRequests` instead, similar to `ResourceLimits`.

File: pkg/scheduler/testing/wrappers.go [hunk 2]
```diff
@@ -619,6 +630,28 @@ func (p *PodWrapper) Req(resMap map[v1.ResourceName]string) *PodWrapper {
 	return p
```
Comment (kerthcet) [inline pkg/scheduler/testing/wrappers.go:630]: Rename this to `Res`, which means add both requests and limits to containers. Add another method `Req` for only requests.

File: pkg/scheduler/extender_test.go [hunk 3]
```diff
@@ -361,6 +361,14 @@ func TestIsInterested(t *testing.T) {
 			}).Obj(),
 			want: true,
 		},
+		{
+			label:    "Managed memory, container memory limit",
```
Comment (kerthcet) [inline pkg/scheduler/extender_test.go:365]: s/limit/with Limits/, also rename case above `Managed memory, container memory with Requests`

File: pkg/scheduler/testing/wrappers.go [hunk 4]
```diff
@@ -187,6 +187,30 @@ func (c *ContainerWrapper) Resources(resMap map[v1.ResourceName]string) *Contain
 	return c
 }
 
+// ResourceRequests sets the container resources to the given resource map.
+func (c *ContainerWrapper) ResourceRequests(resMap map[v1.ResourceName]string) *ContainerWrapper {
```
Comment (kerthcet) [inline pkg/scheduler/testing/wrappers.go:191]: s/resMap/reqMap/

File: pkg/scheduler/testing/wrappers.go [hunk 5]
```diff
@@ -187,6 +187,30 @@ func (c *ContainerWrapper) Resources(resMap map[v1.ResourceName]string) *Contain
 	return c
 }
 
+// ResourceRequests sets the container resources to the given resource map.
```
Comment (kerthcet) [inline pkg/scheduler/testing/wrappers.go:190]: sets the container resource requests

File: pkg/scheduler/testing/wrappers.go [hunk 6]
```diff
@@ -608,14 +632,36 @@ func (p *PodWrapper) Annotations(annotations map[string]string) *PodWrapper {
...
+	if len(resMap) == 0 {
+		return p
+	}
+
+	name := fmt.Sprintf("con%d", len(p.Spec.Containers))
+	p.Spec.Containers = append(p.Spec.Containers, MakeContainer().Name(name).Image(imageutils.GetPauseImageName()).Resources(resMap).Obj())
+	return p
+}
+
 // Req adds a new container to the inner pod with given resource map.
```
Comment (kerthcet) [inline pkg/scheduler/testing/wrappers.go:646]: with given resource map of requests.

File: pkg/scheduler/testing/wrappers.go [hunk 7]
```diff
@@ -608,14 +632,36 @@ func (p *PodWrapper) Annotations(annotations map[string]string) *PodWrapper {
...
 		return p
 	}
 
 	name := fmt.Sprintf("con%d", len(p.Spec.Containers))
-	p.Spec.Containers = append(p.Spec.Containers, MakeContainer().Name(name).Image(imageutils.GetPauseImageName()).Resources(resMap).Obj())
+	p.Spec.Containers = append(p.Spec.Containers, MakeContainer().Name(name).Image(imageutils.GetPauseImageName()).ResourceRequests(resMap).Obj())
+	return p
+}
+
+// Lim adds a new container to the inner pod with given resource limit map.
```
Comment (kerthcet) [inline pkg/scheduler/testing/wrappers.go:657]: with given resource map of limits.

File: pkg/scheduler/testing/wrappers.go [hunk 8]
```diff
@@ -608,14 +632,36 @@ func (p *PodWrapper) Annotations(annotations map[string]string) *PodWrapper {
...
+
 // Req adds a new container to the inner pod with given resource map.
 func (p *PodWrapper) Req(resMap map[v1.ResourceName]string) *PodWrapper {
 	if len(resMap) == 0 {
 		return p
 	}
 
 	name := fmt.Sprintf("con%d", len(p.Spec.Containers))
-	p.Spec.Containers = append(p.Spec.Containers, MakeContainer().Name(name).Image(imageutils.GetPauseImageName()).Resources(resMap).Obj())
+	p.Spec.Containers = append(p.Spec.Containers, MakeContainer().Name(name).Image(imageutils.GetPauseImageName()).ResourceRequests(resMap).Obj())
```
Comment (kerthcet) [inline pkg/scheduler/testing/wrappers.go:653]: Rename resMap to reqMap, short for requestsMap.

File: test/integration/scheduler/scoring/priorities_test.go [hunk 9]
```diff
@@ -151,23 +151,27 @@ func TestNodeResourcesScoring(t *testing.T) {
...
 	}
-	cpuBoundPod1, err := runPausePod(testCtx.ClientSet, st.MakePod().Namespace(testCtx.NS.Name).Name("cpubound1").Req(
-		map[v1.ResourceName]string{
-			v1.ResourceCPU:    "2",
-			v1.ResourceMemory: "4G",
-			resourceGPU:       "1",
-		},
-	).Obj())
+
+	cpuBoundResourceMap := map[v1.ResourceName]string{
```
Comment (kerthcet) [inline test/integration/scheduler/scoring/priorities_test.go:155]: If we want to reuse this variable through the test cases, then better to add this in the head of this function. But I suggest to revert this change, little improvements, and we forgot L146-L149, code style is inconsistent, WDYT?

Comment (sathyanarays) [reply test/integration/scheduler/scoring/priorities_test.go]: Reverted the variable changes.

File: pkg/scheduler/testing/wrappers.go (hunk 8 above)
Comment (sathyanarays) [reply pkg/scheduler/testing/wrappers.go]: Done

File: pkg/scheduler/testing/wrappers.go (hunk 7 above)
Comment (sathyanarays) [reply pkg/scheduler/testing/wrappers.go]: Done

**Comment by kerthcet** [comment]
/lgtm

Thanks @sathyanarays , welcome to the community!
 Leave another review and approval to @alculquicondor

**Review by alculquicondor — COMMENTED** [review]
/approve
